### PR TITLE
AER-2735 Link Habitat or Species to a custom assessment point

### PR DESCRIPTION
**Pull request by MichielJanssen-DAT**
- Updated NcaCustomCalculationPoints to include a list of entityReferences
- Updated JSON test file (tested in JsonRoundTripTest) to also include this new entityReferences

### CODE REVIEW

File: source/imaer-shared/src/main/java/nl/overheid/aerius/shared/domain/v2/point/EntityType.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+package nl.overheid.aerius.shared.domain.v2.point;
+
+/**
+ * Entity types of entity references
+ */
+public enum EntityType {
+  HABITAT,
```
Comment (BertScholten) [inline source/imaer-shared/src/main/java/nl/overheid/aerius/shared/domain/v2/point/EntityType.java:23]: Would prefer `HABITAT_TYPE`: just habitat could also be a habitat area, and that's not what we're referencing).

File: source/imaer-shared/src/main/java/nl/overheid/aerius/shared/domain/v2/point/EntityReference.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+package nl.overheid.aerius.shared.domain.v2.point;
+
+public class EntityReference {
```
Comment (BertScholten) [inline source/imaer-shared/src/main/java/nl/overheid/aerius/shared/domain/v2/point/EntityReference.java:19]: We've been adding some javadoc on class level recently (see `StandardColdStartVehicles` and `CustomAdditionalHousingSystem`), might be a good idea to add a brief description here as well

File: source/imaer-shared/src/main/java/nl/overheid/aerius/shared/domain/v2/point/NcaCustomCalculationPoint.java [hunk 3]
```diff
@@ -24,6 +26,7 @@ public class NcaCustomCalculationPoint extends CustomCalculationPoint {
   private static final long serialVersionUID = 1L;
 
   private Double roadLocalFractionNO2;
+  private List<EntityReference> entityReferences;
```
Comment (BertScholten) [inline source/imaer-shared/src/main/java/nl/overheid/aerius/shared/domain/v2/point/NcaCustomCalculationPoint.java:29]: Generally we already initialize collections in `domain.v2`, in the case of lists usually with an `ArrayList`.

File: source/imaer-shared/src/main/java/nl/overheid/aerius/shared/domain/v2/point/NcaCustomCalculationPoint.java [hunk 4]
```diff
@@ -33,8 +36,17 @@ public void setRoadLocalFractionNO2(final Double roadLocalFractionNO2) {
     this.roadLocalFractionNO2 = roadLocalFractionNO2;
   }
 
+  public List<EntityReference> getEntityReferences() {
+    return entityReferences;
+  }
+
+  public void setEntityReferences(final List<EntityReference> entityReferences) {
+    this.entityReferences = entityReferences;
+  }
```
Comment (BertScholten) [inline source/imaer-shared/src/main/java/nl/overheid/aerius/shared/domain/v2/point/NcaCustomCalculationPoint.java:45]: Should have mentioned this earlier, but there is a `JsonRoundTripTest` that tests if JSON (de)serializaton works in this module. You could add a new test case for this part (or add it to the existing `NcaCustomCalculationPoint.json` file).

**Review by BertScholten — CHANGES_REQUESTED** [review]

File: source/imaer-shared/src/main/java/nl/overheid/aerius/shared/domain/v2/point/EntityType.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+package nl.overheid.aerius.shared.domain.v2.point;
+
+/**
+ * Entity types of entity references
+ */
+public enum EntityType {
+  HABITAT_TYPE,
+  SPECIES_TYPE
```
[the remark this answers is not in the document]
Comment (MichielJanssen-DAT) [reply source/imaer-shared/src/main/java/nl/overheid/aerius/shared/domain/v2/point/EntityType.java]: Alright, we can always add it back in the future along with other possible entity types if needed

**Review by BertScholten — APPROVED** [review]
LGTM

**Pull request merged**